flutter/android/src/main/kotlin/io/sentry/flutter/SentryFlutterPlugin.kt

```diff
@@ -66,36 +66,57 @@ class SentryFlutterPlugin : FlutterPlugin, MethodCallHandler {
-      options.environment = args["environment"] as String?
-      options.release = args["release"] as String?
-      options.dist = args["dist"] as String?
-      options.isEnableSessionTracking = args["enableAutoSessionTracking"] as Boolean
-      options.sessionTrackingIntervalMillis = (args["autoSessionTrackingIntervalMillis"] as Int).toLong()
-      options.anrTimeoutIntervalMillis = (args["anrTimeoutIntervalMillis"] as Int).toLong()
-      // expose options for isAttachThreads?
-      options.isAttachStacktrace = args["attachStacktrace"] as Boolean
-
-      val enableAutoNativeBreadcrumbs = args["enableAutoNativeBreadcrumbs"] as Boolean
-      options.isEnableActivityLifecycleBreadcrumbs = enableAutoNativeBreadcrumbs
-      options.isEnableAppLifecycleBreadcrumbs = enableAutoNativeBreadcrumbs
-      options.isEnableSystemEventBreadcrumbs = enableAutoNativeBreadcrumbs
-      options.isEnableAppComponentBreadcrumbs = enableAutoNativeBreadcrumbs
-
-      options.maxBreadcrumbs = args["maxBreadcrumbs"] as Int
-      options.cacheDirSize = args["cacheDirSize"] as Int
-
-      val level = args["diagnosticLevel"] as String
-      val sentryLevel = SentryLevel.valueOf(level.toUpperCase(Locale.ROOT))
-      options.setDiagnosticLevel(sentryLevel)
-
-      val anrEnabled = args["anrEnabled"] as Boolean
+      (args["dsn"] as? String)?.let {
+        options.dsn = it
+      }
+      (args["debug"] as? Boolean)?.let {
+        options.isDebug = it
+      }
+      (args["environment"] as? String)?.let {
+        options.environment = it
+      }
+      (args["release"] as? String)?.let {
+        options.release = it
+      }
+      (args["dist"] as? String)?.let {
+        options.dist = it
+      }
+      (args["enableAutoSessionTracking"] as? Boolean)?.let {
+        options.isEnableSessionTracking = it
+      }
+      (args["autoSessionTrackingIntervalMillis"] as? Long)?.let {
+        options.sessionTrackingIntervalMillis = it
+      }
+      (args["anrTimeoutIntervalMillis"] as? Long)?.let {
+        options.anrTimeoutIntervalMillis = it
+      }
+      (args["attachThreads"] as? Boolean)?.let {
+        options.isAttachThreads = it
+      }
+      (args["attachStacktrace"] as? Boolean)?.let {
+        options.isAttachStacktrace = it
+      }
+      (args["enableAutoNativeBreadcrumbs"] as? Boolean)?.let {
+        options.isEnableActivityLifecycleBreadcrumbs = it
+        options.isEnableAppLifecycleBreadcrumbs = it
+        options.isEnableSystemEventBreadcrumbs = it
+        options.isEnableAppComponentBreadcrumbs = it
+      }
+      (args["maxBreadcrumbs"] as? Int)?.let {
+        options.maxBreadcrumbs = it
+      }
+      (args["cacheDirSize"] as? Int)?.let {
+        options.cacheDirSize = it
+      }
+      (args["diagnosticLevel"] as? String)?.takeUnless { options.isDebug == false }?.let {
+        val sentryLevel = SentryLevel.valueOf(it.toUpperCase(Locale.ROOT))
+        options.setDiagnosticLevel(sentryLevel)
+      }
```

Comment:
what's about just doing this inside of the:

```
(args["debug"] as? Boolean)?.let {
      options.isDebug = it
      if (options.isDebug){
        // read diagnosticLevel
      }
}
```

in this case, we don't even check if `diagnosticLevel` exists or do `takeUnless`, so we save some allocation.

Reply:
Wouldn't we then be setting it only if `options.debug` is true from the flutter side, and not set it if the native default `options.debug` is set to true? Does this matter?

Reply:
good catch, it makes sense in both cases :)